apps/opik-documentation/documentation/fern/docs/evaluation/evaluate_your_llm.mdx

```diff
@@ -626,6 +626,175 @@ evaluation = evaluate(
 )
 ```
 
+### Using task span evaluation metrics
+
+Opik supports advanced evaluation metrics that can analyze the detailed execution information of your LLM tasks. These metrics receive a `task_span` parameter containing structured data about the task execution, including input, output, metadata, and nested operations.
+
+Task span metrics are particularly useful for evaluating:
+- The internal structure and behavior of your LLM applications
+- Performance characteristics like execution patterns
+- Quality of intermediate steps in complex workflows
+- Cost and usage optimization opportunities
+- Agent trajectory
+
+#### Creating task span metrics
+
+To create a task span evaluation metric, define a metric class that accepts a `task_span` parameter in its `score` method. The `task_span` parameter is a [`SpanModel`](https://www.comet.com/docs/opik/python-sdk-reference/message_processing_emulation/SpanModel.html) object that contains detailed information about the task execution:
+
+<CodeBlocks>
+```python title="Python" language="python"
+from typing import Optional
+from opik.evaluation.metrics import BaseMetric, score_result
+from opik.message_processing.emulation.models import SpanModel
+
+class TaskSpanQualityMetric(BaseMetric):
```

Comment:
Let's maybe just keep the latency metric, the task span quality is quite hard to understand.
Or change it to something simpler, e.g. check if there are errors recursively, or find the total tokens amount.